ext/pdo_sqlite/sqlite_driver.c

```diff
@@ -45,33 +45,33 @@ int _pdo_sqlite_error(pdo_dbh_t *dbh, pdo_stmt_t *stmt, const char *file, int li
 		}
 		einfo->errmsg = pestrdup((char*)sqlite3_errmsg(H->db), dbh->is_persistent);
 	} else { /* no error */
-		strncpy(*pdo_err, PDO_ERR_NONE, sizeof(PDO_ERR_NONE));
+		strlcpy(*pdo_err, PDO_ERR_NONE, sizeof(PDO_ERR_NONE));
```

Comment:
This doesn't fix the issue reported in the warning, the used size is still that of the source, not the destination. The sizeof(PDO_ERR_NONE) here should be replaced by sizeof(*pdo_err) (assuming that correctly returns 6).